.agent/docs/conventions_and_patterns.md

```
@@ -0,0 +1,196 @@
+const factor = new AppModuleBuilder();
+factor
+  .use(...FunctionalityModules)       // Always loaded
+  .use(AuthModule, PermissionModule)  // Always loaded
+  .useIf(() => env.flavors.graphql, GqlModule, PaymentModule, ...)
+  .useIf(() => env.flavors.sync, SyncModule)
+  .useIf(() => env.flavors.doc, DocServiceModule)
+  .compile();
+```
+
+**Server Flavors:**
+| Flavor | Purpose |
```

markdownlint-cli2 (0.20.0)

[warning] Tables should be surrounded by blank lines

(MD058, blanks-around-tables)